### DISCUSSION TITLE
Adding metadata to ORDER

### DISCUSSION THREAD

**Discussion by aetherfaerber** [kitodo/kitodo-production#6299] [Q&A] [answered]
Currently, the content of ORDERLABEL can be changed through the metadata editor's pagination feature with various options. Following the different uses of LABEL and ORDERLABEL as defined in [the code](https://github.com/kitodo/kitodo-production/blob/d074f6182815406e16bd4ed4122c0c021c3f8577/Kitodo-API/src/main/java/org/kitodo/api/dataformat/Division.java#L51) this seems to be a sensible choice.

OTOH, the ”rename files“ feature strictly uses ORDER for renaming of files. This also seems like a sensible choice. But the only available configuration option for this feature is the length of the resulting name, i.e. number of leading zeros. But we need our files to have identifiable and human readable names. Also, in some cases, we would prefer pagination-like naming of files, e.g. ”r“ and ”v“ for deeds instead of numbers. But we cannot add any custom metadata to the filenames, at least not inside the metadata editor.
Having an automatic step after closing the metadata editor that reads the metadata from the xml and adds it as prefix to the filenames is not complicated to implement but duplicates the process of renaming all files which is fairly inefficient.

One option is to have the features available now for pagination (where we probably wont need the advanced options) available for renaming. This could be realized through a switch in the pagination editor to either change the values assigned to ORDER or to ORDERLABEL. 
But I guess this could be confusing to users who don't care about filenames.

Another option to would be to define some metadata fields in the project details that should always prefix filenames or define these fields in the ruleset to be added to ORDER.

How does everyone else deal with this?

**Comment by henning-gerhardt** [comment, marked as answer]
We at SLUB have really strict file name rules and the file names are ordered in the ordering of the physical original work. Renaming could only be done in a specific task and with more work just renaming / adjusting the ordering or adding or removing media files as there are more rules for our files and file content which must be proofed before remaining work can be done. This is not needed for displaying inside meta data editor but for presentation and ingesting them to our long term preservation system.

**Reply by aetherfaerber** [reply]
Thank you very much for your swift reply! Of course, our file name rules are also in place for further use in presentation and preservation systems (although in our case the folder structure is the crucial part and filenames could be theoretically neglected). You are right, the possibility to generate the filenames based on the pagination editor bears the risk of malformed filenames.
I don't think this risk exists if the metadata fields that should be entered as prefix in the filenames would be defined outside the metadata editor. Anyway, in that case there is still a risk that the person using the metadata editor omits or forgets to click the button for renaming. As you pointed out, a separate task is the failure proof way to go.
